<?xml version="1.0"?>
<launch>
    <!-- Launch the planning node with obstacle visualization -->
    <node name="planning_node_with_visualization" 
          pkg="RVizVisualization" 
          type="planning_node_with_visualization" 
          output="screen">
        
        <!-- Robot and environment parameters -->
        <param name="robot_description_path" value="$(find RVizVisualization)/../../res/panda.urdf"/>
        <param name="environment_path" value="$(find RVizVisualization)/../../res/obstacles.xml"/>
        
        <!-- Visualization parameters -->
        <param name="obstacle_frame_id" value="base_link"/>
        <param name="obstacle_topic" value="/planning_obstacles"/>
        <param name="visualization_rate" value="2.0"/>
    </node>
    
    <!-- Launch RViz with our custom configuration -->
    <node name="rviz" 
          pkg="rviz" 
          type="rviz" 
          args="-d $(find RVizVisualization)/config/planning_visualization.rviz"
          output="screen"/>
    
    <!-- Optional: Launch a static transform publisher for the base frame -->
    <node pkg="tf2_ros" 
          type="static_transform_publisher" 
          name="base_link_publisher" 
          args="0 0 0 0 0 0 map base_link"/>

</launch>
